### PR TITLE
Bugfix closing the datepicker should cancel the ongoing selection 1836

### LINKED ISSUE #1836: Bugfix: Closing the datepicker should cancel the ongoing selection
### Overview

We need to provide a way to cancel and reset the date selection so that users aren't stuck in an unspecified UX flow.

### Action Items

- [ ] Add state to track the selected dates of the datepicker that exist prior to the user selecting a new Start Date
  - e.g. initialStartDate, initialEndDate
- [ ] Provide an event handler that restores the initial dates when the user clicks out of the datepicker dropdown, essentially canceling the datepicker flow
  - the map should not load data for the new dates, this cancellation action should be [idempotent](https://developer.mozilla.org/en-US/docs/Glossary/Idempotent)

### Dev Resources

Useful Files and Components
- Map.jsx
- FilterMenu.jsx
- DateSelector.jsx

### Resources/Instructions

- [Bug Report: (see Steps to Reproduce)](https://github.com/hackforla/311-data/issues/1726#issuecomment-2359698327)

### PR DESCRIPTION
**Pull request by DorianDeptuch**
Fixes #1836 

  - [ ] Up to date with `main` branch
  - [ ] Branch name follows [guidelines](https://github.com/hackforla/311-data/blob/master/GETTING_STARTED.md#feature-branching)
  - [ ] All PR Status checks are successful
  - [ ] Peer reviewed and approved

Any questions? See the [getting started guide](https://github.com/hackforla/311-data/blob/master/GETTING_STARTED.md)

### CODE REVIEW

File: components/common/ReactDayPicker/ReactDayPicker.jsx [hunk 1]
```diff
@@ -175,13 +170,16 @@ function ReactDayPicker({
...
-
-    // Our date range selection logic is very simple: the user is selecting the
-    // first day in their date range if from and to are set, or if they're both
-    // unset. Otherwise, they are selecting the last day.
-    if (!(startDate && endDate)) {
-      // If the user picks the first date then picks the second date that is before the first date
-      // Reassign the From and To Day
+    
+  //If both startDate and endDate were already selected. Start a new range selection.
+  if(startDate && endDate){
```
Comment (kdow) [inline components/common/ReactDayPicker/ReactDayPicker.jsx:175]: nit: add space after `if` (lines 175 and 180)

File: components/common/ReactDayPicker/ReactDayPicker.jsx [hunk 2]
```diff
@@ -175,13 +170,16 @@ function ReactDayPicker({
...
+    setFromDay(day);
+    updateEndDate(null);
+    setEnteredTo(null);
+    //If startDate is selected and endDate is unselected, complete the range selection.
```
Comment (kdow) [inline components/common/ReactDayPicker/ReactDayPicker.jsx:179]: nit: for comments, add space between `//` and start of comment

**Review by kdow — COMMENTED** [review]
I verified this fix works. If any other changes are needed, please include my suggested changes with that update.

**Review by efrenmarin45 — APPROVED** [review]
Reviewed the changes and it does resolve the issue brought by #1836

LGTM

**Review by ryanfchase — APPROVED** [review]
@DorianDeptuch feel free to make the "nit" adjustments, otherwise this is looking good 👍

**Review by ryanfchase — APPROVED** [review]
Approved. We're ready to punch this through 👍

Merge when ready

**Pull request merged**